security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -839,12 +850,15 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token string, connKey
 
 	// If token is jwt format, construct host name from jwt with format like spiffe://cluster.local/ns/foo/sa/sleep
 	// otherwise just use sdsrequest.resourceName as csr host name.
-	csrHostName, err := constructCSRHostName(sc.configOptions.TrustDomain, token)
-	if err != nil {
-		cacheLog.Warnf("%s failed to extract host name from jwt: %v, fallback to SDS request"+
-			" resource name: %s", logPrefix, err, connKey.ResourceName)
-		csrHostName = connKey.ResourceName
+	csrHostName := connKey.ResourceName
+	if sc.secOpts.CredFetcher.GetType() == security.K8S {
```

Comment:
This seems like a leaky abstraction that the credential fetcher has leaked into the secret cache code

Reply:
See my other comments. I want to keep the existing behavior on k8s platform to be safe. We may clean this up later.

Reply:
To keep it safe - don't change it... Reading a token from a file is not a k8s platform specific behavior, and just like loading certs from files should be the simplest and safest code path, without over-complexity.